pkgs/applications/blockchains/lighthouse/default.nix

```
@@ -0,0 +1,76 @@
+{ lib, stdenv
+, fetchFromGitHub, fetchgit
+, rustPlatform
+, pkg-config, cmake, protobuf
+, zlib, bzip2, openssl, leveldb
+, nodePackages, curl, gnumake, git
+, portable ? true
+}:
+
+let
+  # Contains the data required by the deposit_contract
+  # See `common/deposit_contract/build.rs` in the lighthous source for exact version
+  eth2specs = fetchFromGitHub {
+    owner = "ethereum";
+    repo = "eth2.0-specs";
+    rev = "v0.12.1";
+    sha256 = "05irxnskcsdbp844hgk2hwkjmgfjxin99jr1fl5b5wg1v7v7gl4y";
+  };
+  eth2specs_unsafe = fetchFromGitHub {
+    owner = "sigp";
+    repo = "unsafe-eth2-deposit-contract";
+    # NOTE: the version of the unsafe contract lags the main tag, but the v0.9.2.1 code is compatible
+    # with the unmodified v0.12.1 contract. See `common/deposit_contract/build.rs` for details.
+    rev = "v0.9.2.1";
+    sha256 = "151ahdj5114xb5hbhzsx5rx4yyry832hhm0bkx36s2np3r8v883q";
+  };
+in
+rustPlatform.buildRustPackage rec {
+  pname = "lighthouse";
+  version = "1.4.0";
+
+  src = fetchgit {
+    url = "https://github.com/sigp/lighthouse.git";
```

Use `fetchFromGitHub`:

```suggestion
  src = fetchFromGitHub {
    owner = "sigp";
    repo = pname;
```

(make sure you change the hash below).

---

`fetchFromGitHub` didn't work as lighthouse requires the git information to determin it's version. Without `leaveDotGit` tests fail.

---

fetchFromGitHub also support leaveDotGit.

---

@SuperSandro2000 @prusnak  I changed this to use `leaveDotGit` with `fetchFromGitHub`